generator/altitude_generator.cpp

```diff
@@ -0,0 +1,247 @@
+#include "generator/routing_generator.hpp"
+#include "generator/srtm_parser.hpp"
+
+#include "routing/routing_helpers.hpp"
+
+#include "indexer/altitude_loader.hpp"
+#include "indexer/feature.hpp"
+#include "indexer/feature_altitude.hpp"
+#include "indexer/feature_data.hpp"
+#include "indexer/feature_processor.hpp"
+
+#include "coding/file_container.hpp"
+#include "coding/file_name_utils.hpp"
+#include "coding/read_write_utils.hpp"
+#include "coding/reader.hpp"
+#include "coding/varint.hpp"
+
+#include "coding/internal/file_data.hpp"
+
+#include "base/assert.hpp"
+#include "base/logging.hpp"
+#include "base/stl_helpers.hpp"
+#include "base/string_utils.hpp"
+
+#include "defines.hpp"
+
+#include "3party/succinct/elias_fano.hpp"
+#include "3party/succinct/mapper.hpp"
+#include "3party/succinct/rs_bit_vector.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/type_traits.hpp"
+#include "std/utility.hpp"
+#include "std/vector.hpp"
+
+using namespace feature;
+
+namespace
+{
+TAltitudeSectionVersion constexpr kAltitudeSectionVersion = 1;
+
+class Processor
+{
+public:
+  using TFeatureAltitude = pair<uint32_t, Altitude>;
+  using TFeatureAltitudes = vector<TFeatureAltitude>;
+
+  Processor(string const & srtmPath) : m_srtmManager(srtmPath), m_minAltitude(kInvalidAltitude) {}
+
+  TFeatureAltitudes const & GetFeatureAltitudes() const { return m_featureAltitudes; }
+
+  vector<bool> const & GetAltitudeAvailability() const { return m_altitudeAvailability; }
+
+  TAltitude GetMinAltitude() const { return m_minAltitude; }
+
+  void operator()(FeatureType const & f, uint32_t const & id)
+  {
+    if (id != m_altitudeAvailability.size())
+    {
+      LOG(LERROR, ("There's a gap in feature id order."));
+      return;
+    }
+
+    bool hasAltitude = false;
+    do
```

Comment:
Please, get rid of this while and use MY_SCOPE_GUARD to push something at the m_altitudeAvailability.